### PR TITLE
MBS-9607: Rewrite MusicBrainz not found pages in React/JSX

### PR DESCRIPTION
**Pull request by spellew**
Replaced all the not_found.tt files with equivalent templates written in React/JSX. Accepted on Google Code-In by yvanzo.

### CODE REVIEW

**Comment by yvanzo** [comment]
Depends on PR https://github.com/metabrainz/musicbrainz-server/pull/599.

**Comment by mwiencek** [comment]
I think we should create a reusable `NotFound` component to reduce some of the duplication here. Here's an example of what that would look like:

```JavaScript
const EditNotFound = () => (
  <NotFound
    title={l('Edit Not Found')}
    message={l('Sorry, we could not find an edit with that edit ID. You may wish to try and perform an {search_url|edit search} instead.', {__react: true, search_url: '/search/edits'})}
  />
);
```

File: root/tag/NotFound.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// later version: http://www.gnu.org/licenses/gpl-2.0.txt
+
+const React = require('react');
+const NotFound = require('../components/NotFound');
+const {l} = require('../static/scripts/common/i18n');
+
+const TagNotFound = () => (
+  <NotFound title={l('Tag Not Used')}>
+    <p>{l('No MusicBrainz entities have yet been tagged with "' + $c.stash.tag + '".')}</p>
+    <p>{l('If you wish to use this tag, please {search_url|search} for the entity first and apply the tag using the sidebar.',
```
Comment (mwiencek) [inline root/tag/NotFound.js]: Please don't change the strings unless necessary. I saw you changed `{url|search}` to `{search_url|search}` and had to go through all the rest to make sure there weren't other changes. These changes require translators to retranslate things for no reason.

(No need to push fixes, I'm taking care of it.)

**Comment by mwiencek** [comment]
Fixed it so that the pages actually still 404 in 9902927.

**Pull request merged**